x-pack/solutions/security/plugins/security_solution/public/detections/components/attacks/table/table_section.test.tsx

```diff
@@ -377,4 +377,55 @@ describe('<TableSection />', () => {
       });
     });
   });
+
+  describe('enforced groups', () => {
+    it('should include enforcedGroups in groupingSettings', () => {
+      expect(groupingSettings.enforcedGroups).toBeDefined();
+      expect(Array.isArray(groupingSettings.enforcedGroups)).toBe(true);
+    });
+
+    it('should enforce ALERT_ATTACK_IDS as the enforced group', () => {
+      expect(groupingSettings.enforcedGroups).toEqual([ALERT_ATTACK_IDS]);
+    });
+
+    it('should pass enforcedGroups setting to GroupedAlertsTable', async () => {
+      render(
+        <TestProviders>
+          <TableSection dataView={dataView} statusFilter={[]} pageFilters={[]} />
+        </TestProviders>
+      );
+
+      await waitFor(() => {
+        expect(GroupedAlertsTable).toHaveBeenCalled();
+        const [props] = (GroupedAlertsTable as unknown as jest.Mock).mock.calls[0];
+        expect(props.settings.enforcedGroups).toEqual([ALERT_ATTACK_IDS]);
+      });
+    });
+
+    it('should pass all grouping settings including enforcedGroups', async () => {
+      render(
+        <TestProviders>
+          <TableSection dataView={dataView} statusFilter={[]} pageFilters={[]} />
+        </TestProviders>
+      );
+
+      await waitFor(() => {
+        expect(GroupedAlertsTable).toHaveBeenCalled();
+        const [props] = (GroupedAlertsTable as unknown as jest.Mock).mock.calls[0];
+        expect(props.settings).toMatchObject({
+          hideNoneOption: true,
+          hideCustomFieldOption: true,
+          hideOptionsTitle: true,
+          enforcedGroups: [ALERT_ATTACK_IDS],
+        });
+      });
+    });
+
+    it('should have enforcedGroups that matches the groupingOptions key', () => {
+      const enforcedGroupKey = groupingSettings.enforcedGroups?.[0];
+      const groupingOptionKey = groupingOptions[0]?.key;
+      expect(enforcedGroupKey).toBe(groupingOptionKey);
+      expect(enforcedGroupKey).toBe(ALERT_ATTACK_IDS);
+    });
```

Comment:
Finally, this also looks like a weird test to me. It feels like it's trying to test the UI test without being a UI test no?

Why not getting the DOM element with `group-selector-dropdown` for `data-test-subj` and verifying that it show `Group by attack`? Isn't the outcome similar?

Reply:
Good point - will remove it!